src/ReasonReact.re

```diff
@@ -883,6 +883,17 @@ module Router = {
     | Some((window: Dom.window)) =>
       removeEventListener(window, "popstate", watcherID)
     };
+
+  let useUrl = () => {
+    let (url, setUrl) = React.useState(dangerouslyGetInitialUrl);
+
+    React.useLayoutEffect0(() => {
```

Comment:
Why did you opt for LayoutEffect here? I think that because this doesn’t need to be synchronous Effect might be a better choice, but I’m not sold either way really. I don’t think this is blocking this PR but since it would technically be a breaking change I think we should resolve before a wider release.

Reply:
The use case I had was only working with `useLayoutEffect`.

(In my app, for some routes, I would render a `Redirect` component that would push a new route. For some reason, this only worked when using `useLayoutEffect` here and `useEffect` in the `Redirect` component. I did not investigate it any further.)

Reply:
Interesting. LayoutEffect happens earlier than Effect so perhaps these hooks are called upward and this order is required in order to trigger this subscription before the Redirect component executes. That seems like a reasonable excuse to put this here.